src/ru/animelib/src/eu/kanade/tachiyomi/animeextension/ru/animelib/Animelib.kt

```diff
@@ -0,0 +1,500 @@
+    override fun searchAnimeParse(response: Response) = popularAnimeParse(response)
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val searchParams = AnimelibFilters.getSearchParameters(filters)
+
+        val url = apiUrl.toHttpUrl().newBuilder()
+        url.addPathSegment("anime")
+        url.addQueryParameter("page", page.toString())
+        url.addQueryParameter("site_id[]", "5")
+        url.addQueryParameter("links[]", "")
+
+        searchParams.genres.include.forEach { url.addQueryParameter("genres[]", it) }
+        searchParams.genres.exclude.forEach { url.addQueryParameter("genres_exclude[]", it) }
+        searchParams.format.forEach { url.addQueryParameter("types[]", it) }
+        searchParams.pegi.forEach { url.addQueryParameter("caution[]", it) }
+        searchParams.ongoingStatus.forEach { url.addQueryParameter("status[]", it) }
+
+        url.addQueryParameter("sort_by", searchParams.sortOrder)
+        if (searchParams.sortDirection.isNotEmpty()) {
+            url.addQueryParameter("sort_type", searchParams.sortDirection)
+        }
+
+        url.addQueryParameter("q", query)
+
+        return GET(url.build())
+    }
+
+    override fun getFilterList() = AnimelibFilters.FILTER_LIST
+
+    // =============================== Utils ===============================
+    private fun fetchPreferredVideoServer(): String {
+        val url = apiUrl.toHttpUrl().newBuilder()
+        url.addPathSegment("constants")
+        url.addQueryParameter("fields[]", "videoServers")
+
+        val videoServerResponse = client.newCall(GET(url.build())).execute()
+        val videoServers = videoServerResponse.parseAs<VideoServerData>()
+
+        val serverPreference = preferences.getString(PREF_SERVER_KEY, PREF_SERVER_ENTRIES[0])
+        if (serverPreference.isNullOrEmpty()) {
+            return videoServers.data.videoServers[0].url
+        }
+
+        for (videoServer in videoServers.data.videoServers) {
+            if (videoServer.label == serverPreference) {
+                return videoServer.url
+            }
+        }
+
+        return videoServers.data.videoServers[0].url
+    }
+
+    private fun kodikVideoLinks(playerUrl: String?, teamName: String): List<Video> {
+        val useKodik = preferences.getBoolean(PREF_USE_KODIK_KEY, PREF_USE_KODIK_DEFAULT)
+        if (playerUrl.isNullOrEmpty() || !useKodik) {
+            return emptyList()
+        }
+
+        val kodikPage = "https:$playerUrl"
+        val headers = Headers.Builder()
+        headers.add("Referer", baseUrl)
+        val kodikPageResponse = client.newCall(GET(kodikPage, headers.build())).execute()
+
+        // Parse form parameters for video link request
+        val page = kodikPageResponse.asJsoup()
+        val urlParams = page.getElementsByTag("script").find {
+            it.html().contains(domain)
+        }?.html() ?: return emptyList()
```

Comment:
```suggestion
        val urlParams = page.selectFirst("script:containsData($domain)")?.data()
            ?: return emptyList()
```
Use `.data()` instead of `.html()` when you want the javascript